js/util.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2017-present, The Visdom Authors
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React, { useEffect, useRef } from 'react';
+
+// custom hook to get previous value of a variable
+function usePrevious(value) {
```

Comment:
So if I understand this correctly, this will return the value last provided to this function? And then the `useEffect` ends up triggering _after_ the return, updating the value? 

Are you able to have more than one `usePrevious` across your application? (is `React` that good at tracing callsites for `useRef` to hook to?) If so that's pretty cool!

Reply:
**Actually, yes!**

I was curious and tested if one could create a chain of `usePrevious`-calls to get the penultimate value or the value before that.

I.e. the following code-snippet did work.
```
  const contentID1 = props.contentID
  const contentID2 = usePrevious(contentID1);
  const contentID3 = usePrevious(contentID2);
```

As `usePrevious` calls `useRef` and `useEffect` internally, i would expect the function to follow the same semantic rules as those. (I mean that it should not be called in a loop, etc.).

Reply:
I'm learning something new every day here too - this is a cool functionality!